source/merge.d.ts

```diff
@@ -3,6 +3,8 @@ import {Except} from './except';
 /**
 Merge two types into a new type. Keys of the second type overrides keys of the first type.
 
+Used in cases where intersection types would result in a prop having `never` – if two different types for the same property are used, intersection types assume it's type is `never`. `MergeDeep` allows for safe intersection of two types and having the second passed in type have precedence, hence resulting in no unwanted `never` prop types.
```

Comment:
I don't think this text makes it immediately clear enough that you're talking about a different type. Use `@see`.